### DISCUSSION TITLE
Vaut-il mieux utiliser le package NPM ou l'API ?

### DISCUSSION THREAD

**Discussion by rodolphebejot** [betagouv/mon-entreprise#2865] [Q&A] [answered]
Bonjour,

Afin d'estimer les charges d'une entreprise il semble qu'il y ait deux solutions:
- utiliser l'endpoint de l'API `/evaluate`
- utiliser les packages modele-social et publicode 

exemple ici: https://mon-entreprise.urssaf.fr/documentation/dirigeant/auto%E2%80%91entrepreneur

Y'a t'il des avantages à utiliser l'un plutôt que l'autre ? 
Je me pencherai plus pour l'utilisation du package puisqu'il permet d'économiser un appel supplémentaire à l'API depuis une app

Merci pour votre réponse et bonne journée à vous !

**Comment by johangirod** [comment, marked as answer]
Bonjour,

Il y a des avantages et inconvénient pour les deux utilisations : 

**Avantages de l'API**
- Mise à jour automatique
- Peut s'utiliser avec d'autres langages que JavaScript

**Avantage du paquet NPM**
- Maitrise de la mise à jour du modèle (pas de risque de changement cassant silencieux, ce qui peut arriver dans l'API si vous utilisez des [règles expérimentales](https://mon-entreprise.urssaf.fr/d%C3%A9veloppeur/api)
- Pas de dépendance à un service externe (dont vous ne maitrisez pas la disponibilité)
- Plus rapide, car évite un appel réseau

J'espère que ces éléments vous permettront d'arrêter votre choix de manière éclairée :)
